jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/list/CaseInstanceListPresenter.java

```diff
@@ -75,10 +95,19 @@ public void createCaseInstance() {
         newCaseInstancePresenter.show();
     }
 
-    protected void refreshData() {
+    private void caseInstancesServiceCall(int currentPage) {
         caseService.call((List<CaseInstanceSummary> cases) -> {
-            view.setCaseInstanceList(cases);
-        }).getCaseInstances(view.getCaseInstanceSearchRequest());
+            visibleCaseInstances.addAll(cases);
+            view.hideLoadButton(cases.size() < PAGE_SIZE);
+            view.setCaseInstanceList(visibleCaseInstances.stream().collect(toList()));
```

Comment:
`visibleCaseInstances` is a list already, there is probably no need to use streams in here.